packages/grafana-data/src/types/panel.ts

```diff
@@ -110,6 +112,15 @@ export interface PanelProps<T = any> {
 
   /** Time range change handler */
   onChangeTimeRange: (timeRange: AbsoluteTimeRange) => void;
+
+  /** Function that renders a custom tooltip */
+  customTooltipFn?: (
```

Comment:
🤔 this is a pretty big hammer... custom tooltip hook for all plugins everywhere -- that is tied to single frame response.

Is there a way to do this that does not change definitions all the way up?  or is the intention to make this general beyond bar chart?

Reply:
For the moment, at least, it's meant to work only for barchart since that's where it's needed. I also felt that it might be overkill unless we want to generalize custom tooltips across panels.

Another way is to add the custom tooltip function as a barchart panel option (although it's not really a panel option), which would isolate it only to barchart but would probably need some modifications to schema generations since I don't recall seeing functions as a data type that can be set on the cue schemas.